components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.observability.handler.metrichandler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.AbstractExtendedSynapseHandler;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.API;
+import org.apache.synapse.rest.RESTConstants;
+import org.apache.synapse.rest.RESTUtils;
+import org.apache.synapse.transport.nhttp.NhttpConstants;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.core.internal.MicroIntegratorBaseConstants;
+import org.wso2.micro.integrator.observability.handler.metrichandler.prometheus.reporter.PrometheusReporter;
+import org.wso2.micro.integrator.observability.handler.util.MetricConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for instrumenting Metrics by wrapping the implementation and
+ * there by allows the implementation to plug into the handler in runtime.
+ */
+public class MetricHandler extends AbstractExtendedSynapseHandler {
+    private static final String METRIC_REPORTER = "metric_reporter";
+    private static final String DELIMITER = "/";
+    private static final String EMPTY = "";
+
+    private MetricReporter metricReporterInstance;
+    private Class loadedMetricClass;
+    private static Log log = LogFactory.getLog(MetricHandler.class);
+
+    private static final String serverPortOffset = System.getProperty(MetricConstants.PORT_OFFSET);
+    private static final String host = System.getProperty(MicroIntegratorBaseConstants.LOCAL_IP_ADDRESS);
+    private static final String port = System.getProperty(MetricConstants.HTTP_PORT);
+    private static final String javaVersion = System.getProperty(MetricConstants.JAVA_VERSION);
+    private static final String javaHome = System.getProperty(MetricConstants.JAVA_HOME);
+
+    @Override
+    public boolean handleServerInit() {
+        getMetricReporter();
```

Comment:
Make this method return the Metric Reporter

Reply:
Fixed

Reply:
I don't see the fix in the source. Have u pushed your changes?